### ISSUE TITLE
Installing msgpack for Visual Studio C++ 2017

### ISSUE THREAD

**Issue by d223chen** [msgpack/msgpack-c#684]
I'd just like to point out that the correct installation process for Visual Studio C++ is to:

1. Add msgpack-c\include to the "Include Directories" under VC++ directories in Configuration Properties
2. Add msgpack-c\include to the "Library Directories" under VC++ directories in Configuration Properties

The installation instructions on the Quickstart doc suggest to only do 1. I got the sample program to run after doing 2 as well.

I think the issue would generalize to other build systems as well - you have to both include the header files and link the libraries in any system.

**Comment by redboltz** [comment]
@d223chen , C++ part of msgpack-c is a header-only library. So no library linking is required.
C part of msgpack-c requires linking a library. Do you mean the C part ?

**Comment by d223chen** [comment]
Yes, I now realize actually that msgpack is a header only library. So I shouldn't have had to add the header to the Library Directories like I said I had to. I cannot reproduce the error anymore, unfortunately. I somehow suspect it was a glitch with my particular configuration, as I had been following [the steps in this Stack Overflow answer](https://stackoverflow.com/a/2655683/8245487) to get Boost to build and work with Visual Studio 2017 on my machine (some libraries needed building indeed). Somehow along the way my programs couldn't load any debugging symbols anymore either, so I highly suspect I just glitched something. Once again, the solution for installing Boost was to use `vcpkg`:

    vcpkg install boost
